lib/common/src/Dom/Document.php

```diff
@@ -390,6 +390,17 @@ public function loadHTMLFragment($source, $options = 0)
 
         $libxml_previous_state = libxml_use_internal_errors(true);
 
+        $options |= LIBXML_COMPACT;
+
+        /*
+         * LIBXML_HTML_NODEFDTD is only available for libxml 2.7.8+.
+         * This should be the case for PHP 5.4+, but some systems seem to compile against a custom libxml version that
+         * is lower than expected.
+         */
+        if (defined('LIBXML_HTML_NODEFDTD')) {
```

Comment:
I assume the `doctype` will be added if this option is not supplied?

Reply:
The normalization seems to already take care of everything.